auditai/cmh.py

```diff
@@ -0,0 +1,125 @@
+from scipy import stats
+from functools import partial
+import numpy as np
+import pandas as pd
+
+def extract_data(df):
+    # utility function for cmh test and odds ratio
+    # df needs to be a 2x2 data frame with columns 'pass' and 'fail'
+    a = float(df.iloc[0]['pass'])
+    d = float(df.iloc[1]['fail'])
+    b = float(df.iloc[0]['fail'])
+    c = float(df.iloc[1]['pass'])
+    t = a + b + c + d
+
+    r_num = (a*d)/(t)
+    r_den = (b*c)/(t)
+    
+    c_num = a - ((a + b)*(a + c))/(t)
+    c_dem = ((a + b)*(c + d)*(a + c)*(b + d))/(t*t*(t - 1))
+    
+    return r_num, r_den, c_num, c_dem
+    
+    
+def cmh_test(dfs):
+    # cochran-mantel-haenszel test (pairwise special case)
+    # dfs is a 2x2xK data frame,
+    #     or a K-deep stack of 2x2 contingency tables
+    #     representing K locations or time frames
+    # each 2x2 contingency table should have 'pass'/'fail' columns
+    #     and group membership denoted 1/0 for rows
+    data = map(extract_data, dfs)
+    c_nums = [val[2] for val in data]
+    c_dems = [val[3] for val in data]
+    c_num = sum(c_nums)**2
+    c_dem = sum(c_dems)
+    cmh = float(c_num)/float(c_dem)
+    
+    pcmh = 1 - stats.chi2.cdf(cmh, 1)
+    return cmh, pcmh
+    
```

Comment:
Can we add an example of these tests in one of our example notebooks? The purpose is to clarify how this test would used in a typical machine learning problem.